### ISSUE TITLE
get rid of net.WriteTable

### ISSUE THREAD

**Issue by uRandomAlex** [Patcher56/PatchProtect#123]
get rid of net.WriteTable as it's inefficient and definitely not lagfree, net.WriteType is poor too, don't use it

**Comment by Domii894** [comment]
Hi,
why do you know, that net.Write<Type> is that inefficient? Any statistics or somewhat?

If these functions are inefficient, what should we use instead? I can remember that some gmod-developers made major changes on net.Write<Type> to make them more efficient.

**Comment by uRandomAlex** [comment]
@Domii894 Internally net.WriteTable uses net.WriteType that writes UInt in addition to value that indicates type of a value (in table). Looking at the code I can say that you pretty much can use other net library functions for networking. It's the biggest bottleneck in this addon as far as I can say.

**Comment by uRandomAlex** [comment]
@Domii894 Oh I just understood you have misunderstood me. There is a net.WriteType function, that writes ANY type you need, but it sends an additional UInt (number) in addition to the value you need to send. net.WriteTable uses it and it has some inefficient things in it too, so it should be avoided.

**Comment by Domii894** [comment]
@uRandomAlex 
No, i got what you said and checked the implementation of net.WriteTable and found out what you were talking of.

I think the most simple fix would be while loops with net.Write<Type> and net.Read<Type>, because almost every table-row we send has the same type.

As I also said to other ppl, I/we am/are very busy at the moment and don't have time to fix this.
It will likely a short and easy fix, but that it too much too (currently don't even have gmod installed :P).
Maybe someone else makes a pull-request or I/we will do it some time later. :)

**Comment by storm37000** [comment]
im pretty sure the sendtable is best for cpu efficiency but less for network, turning the table into json, compressing, sending the compressed data, decompressing, and un json on the other side would be more cpu intensive but save network. also looping would be probably the worst on the cpu.
